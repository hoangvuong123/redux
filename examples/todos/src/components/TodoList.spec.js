import React from 'react'
import { shallow } from 'enzyme';
import TodoList from './TodoList'

const setup = () => {
  let todos= [{
    id: 1,
    completed: false,
    text: 'todo 1'
  },{
    id: 2,
    completed: true,
    text: 'todo 2'
  }]

  const wrapper = shallow(<TodoList todos={todos}/>);
  return { wrapper };
};

describe('TodoList component', () => {
  it('Should render TodoList component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })
})
