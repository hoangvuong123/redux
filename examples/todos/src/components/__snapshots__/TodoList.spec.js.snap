// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Should render TodoList component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todos={
        Array [
            Object {
              "completed": false,
              "id": 1,
              "text": "todo 1",
            },
            Object {
              "completed": true,
              "id": 2,
              "text": "todo 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Todo
          completed={false}
          id={1}
          onClick={[Function]}
          text="todo 1"
/>,
        <Todo
          completed={true}
          id={2}
          onClick={[Function]}
          text="todo 2"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "completed": false,
          "id": 1,
          "onClick": [Function],
          "text": "todo 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "completed": true,
          "id": 2,
          "onClick": [Function],
          "text": "todo 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Todo
            completed={false}
            id={1}
            onClick={[Function]}
            text="todo 1"
/>,
          <Todo
            completed={true}
            id={2}
            onClick={[Function]}
            text="todo 2"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "completed": false,
            "id": 1,
            "onClick": [Function],
            "text": "todo 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "completed": true,
            "id": 2,
            "onClick": [Function],
            "text": "todo 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
